### ISSUE TITLE
sumologic_collector_ingest_budget_assignment requires hidden field _budget to be set on sumologic_collector

### ISSUE THREAD

**Issue by jmoberly** [SumoLogic/sumologic-terraform-provider#98]
I created budgets and collectors and then used **sumologic_collector_ingest_budget_assignment** to do the association.

After the initial successful Terraform Apply I would see this type of change in subsequent plans. 
```
~ sumologic_collector.collector
       fields.%:       "1" => "0"
       fields._budget: "mystuff-staging" => ""
```

If I apply the above change the next plan fails to find the collector id
**_sumologic_collector_ingest_budget_assignment.assignment: unable to find collector 1234567890 for ingest budget._**

I worked around the issue by adding the hidden **_budget** field to my collector
```
resource "sumologic_collector" "collector" {
  name     = "mystuff-${terraform.workspace}"
  timezone = "UTC"
  fields = {
    _budget = "mystuff-${local.environment}"
  }
}
```

The work-around is not obvious so it would be nice if **sumologic_collector** ignored the hidden fields.

**Comment by DanWaltman** [comment]
I am just starting to use the sumologic terraform provider.  If the collector ignores the hidden field _budget, then there is no way (that I have found) to assign an existing budget to the collector.

**Comment by seansain** [comment]
Because of this issue we have decided to deprecate `sumologic_collector_ingest_budget_assignment` resource in favor of the `_budgets` field. We have added this to the docs as well to make it clear. Fix is in #135 - closing
